File: Projet_LO52/lo52_project_v2/app/src/main/java/com/example/lo52_project_v2/model/dao/AppDataBase.java
package com.example.lo52_project_v2.model.dao;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.example.lo52_project_v2.model.bo.Course;
import com.example.lo52_project_v2.model.bo.Participant;
import com.example.lo52_project_v2.model.bo.Equipe;
import com.example.lo52_project_v2.model.bo.ParticipantEquipe;
import com.example.lo52_project_v2.model.bo.Tour;

@Database(entities = {Participant.class, Equipe.class, Course.class, Tour.class, ParticipantEquipe.class}, exportSchema = false, version = 1)
public abstract class AppDataBase extends RoomDatabase {
    private static AppDataBase instance;

    public static synchronized AppDataBase getInstance(Context context){
        if(instance == null){
            instance = Room.databaseBuilder(context.getApplicationContext(),
                    AppDataBase.class, "CODEP25F1").allowMainThreadQueries()
                    .build();
        }
        return instance;
    }

    public abstract ParticipantDao participantDao();
    public abstract EquipeDao equipeDao();
    public abstract CourseDao courseDao();
    public abstract TourDao tourDao();
    public abstract ParticipantEquipeDao participantEquipeDao();
}

File: TP4/CourseCODEP25/app/src/main/java/com/utbm/nouassi/manou/coursecodep25/database/AppDatabase.kt
package com.utbm.nouassi.manou.coursecodep25.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.utbm.nouassi.manou.coursecodep25.database.dao.CoureurDao
import com.utbm.nouassi.manou.coursecodep25.database.dao.CourseDao
import com.utbm.nouassi.manou.coursecodep25.database.dao.EquipeDao
import com.utbm.nouassi.manou.coursecodep25.database.dao.ParticipantDao
import com.utbm.nouassi.manou.coursecodep25.database.entity.*

@Database(entities = arrayOf(Coureur::class, Course::class, Equipe::class, Participant::class), version = 1)
abstract class AppDatabase : RoomDatabase() {
    abstract fun courseDao(): CourseDao
    abstract fun equipeDao(): EquipeDao
    abstract fun participantDao(): ParticipantDao
    abstract fun coureurDao(): CoureurDao

    companion object {

        @Volatile
        private var INSTANCE: AppDatabase? = null

        fun getInstance(context: Context): AppDatabase {
            synchronized(this) {
                var instance = INSTANCE

                if (instance == null) {
                    instance = Room.databaseBuilder(
                        context.applicationContext,
                        AppDatabase::class.java,
                        "db-course-codep25"
                    )
                        .fallbackToDestructiveMigration()
                        .build()
                    INSTANCE = instance
                }
                return instance
            }
        }
    }
}